src/Symfony/Component/Translation/Dumper/XliffFileDumper.php

```diff
@@ -40,6 +50,21 @@ public function dump(MessageCatalogue $messages, $options = array())
     }
 
     /**
+     * Set XLIFF tool element metadata.
+     *
+     * @link http://docs.oasis-open.org/xliff/v1.2/os/xliff-core.html#tool_elem
+     *
+     * @param array attributes supported by the <tool> element
+     */
+    protected function setToolMetadata( array $meta ){
```

Comment:
@OskarStark The purpose is for translation software when presented with a file to know that Symfony generated it. Knowing this means the software can process it differently to how it might by default. I could elaborate on that, but in general if a file format provides a generator declaration, I see no reason not to use it. Also other projects using the Translation component may have their own requirement for declaring themselves.

@stof making the method private is fine for the Symfony case, but I was under the impression that Components were supposed to be portable. As shown in the test case, a supposed `XliffExtendedFileDumper` sets its own tool signature, hence why I made it protected.